tls/s2n_tls13.h

```diff
@@ -17,14 +17,21 @@
 
 #include "api/s2n.h"
 #include "tls/s2n_crypto.h"
+#include "utils/s2n_compiler.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-S2N_API
-__attribute__((deprecated("The use of TLS1.3 is configured through security policies")))
-extern int s2n_enable_tls13();
+#if S2N_GCC_VERSION_AT_LEAST(4, 5, 0)
+    S2N_API
+    __attribute__((deprecated("The use of TLS1.3 is configured through security policies")))
```

Comment:
Is this message worth the if/else? Should we just never display the reason for the deprecation?

Also, does any line other than the deprecation attribute need to be inside the if/else? Why duplicate the function definition?

Reply:
It seemed better to keep the message instead of deprecating `s2_enable_tls13` without explanation.

The reason I wrapped everything in the `#if`was readability. The following seemed more complex

```c
S2N_API
#if S2N_GCC_VERSION_AT_LEAST(4, 5, 0)
    __attribute__((deprecated("The use of TLS1.3 is configured through security policies")))
#else
    __attribute__((deprecated))
#endif
extern int s2n_enable_tls13();
```